server/src/test/java/org/elasticsearch/index/IndexTests.java

```diff
@@ -10,21 +10,43 @@
 
 import org.elasticsearch.cluster.ClusterState;
 import org.elasticsearch.common.UUIDs;
-import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.test.AbstractXContentSerializingTestCase;
 import org.elasticsearch.test.ESTestCase;
-import org.elasticsearch.xcontent.ToXContent;
-import org.elasticsearch.xcontent.XContentBuilder;
 import org.elasticsearch.xcontent.XContentParser;
-import org.elasticsearch.xcontent.json.JsonXContent;
 
 import java.io.IOException;
 
 import static org.apache.lucene.tests.util.TestUtil.randomSimpleString;
 import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.not;
 
-public class IndexTests extends ESTestCase {
+public class IndexTests extends AbstractXContentSerializingTestCase<Index> {
```

Comment:
IMO this makes more sense as an `AbstractWireSerializingTestCase` - we don't read these things out of XContent very much now that the HLRC is no more.

Reply:
This is still used in org.elasticsearch.cluster.metadata.DataStream#fromXContent. Should we still test it as long as it exists even if it is deprecated?

Reply:
Ah sorry I thought `AbstractXContentSerializingTestCase` and `AbstractWireSerializingTestCase` were mutually exclusive, but now I see that `AbstractXContentSerializingTestCase` inherits from `AbstractWireSerializingTestCase`. This is good then.